packages/integrations/remix/src/index.ts

```diff
@@ -0,0 +1,437 @@
+                  }
+                  window._DMNO_PUBLIC_DYNAMIC_CONFIG = JSON.parse(request.responseText);
+                  
+                  console.log('loaded public dynamic config', window._DMNO_PUBLIC_DYNAMIC_CONFIG);
+                }
+                
+                if (key in window._DMNO_PUBLIC_DYNAMIC_CONFIG) {
+                  return window._DMNO_PUBLIC_DYNAMIC_CONFIG[key];
+                }
+          ` : `
+                if (${JSON.stringify(dmnoInjectionResult.publicDynamicKeys)}.includes(key)) {
+                  throw new Error(\`❌ Unable to access dynamic config item \\\`\${key}\\\` in Astro "static" output mode\`);
+                }
+          `,
+
+          // in dev mode, we'll give a more detailed error message, letting the user know if they tried to access a sensitive or non-existant item
+          isDevMode ? `
+                if (${JSON.stringify(dmnoInjectionResult.sensitiveKeys)}.includes(key)) {
+                  throw new Error(\`❌ \\\`DMNO_PUBLIC_CONFIG.\${key}\\\` not found - it is sensitive and must be accessed via DMNO_CONFIG on the server only\`);
+                } else {
+                  throw new Error(\`❌ \\\`DMNO_PUBLIC_CONFIG.\${key}\\\` not found - it does not exist in your config schema\`);  
+                }
+          ` : ` 
+                throw new Error(\`❌ \\\`DMNO_PUBLIC_CONFIG.\${key}\\\` not found - it may be sensitive or it may not exist at all\`);
+          `,
+          `
+              }
+            });
+          `,
+
+          // DMNO_CONFIG proxy object just to give a helpful error message
+          // TODO: we could make this a warning instead? because it does get replaced during the build and doesn't actually harm anything
+          `
+            window.DMNO_CONFIG = new Proxy({}, {
+              get(o, key) {
+                throw new Error(\`❌ You cannot access DMNO_CONFIG on the client, try DMNO_PUBLIC_CONFIG.\${key} instead \`);
+              }
+            });
+          `,
+
+          // original source
+          src,
+        ].join('\n');
+      }
+
+      return src;
+    },
+
+    // handleHotUpdate({ file, server }) {
+    //   console.log('hot update', file);
+    // },
+  };
+}
+
+
+export function dmnoRemixPreset(): Preset {
+  return {
+    name: 'dmno-remix-preset',
+    remixConfig: async (args) => {
+      // console.log('remixConfig!', args);
+      return {
+        // inject public dynamic config endpoint
+        routes(defineRoutes) {
+          return defineRoutes((route) => {
+            // TODO: ideally we would use a virtual module but remix seems to be making some assumptions about a real file existing
+
+            // instead awkwardly we need to use a relative path and need it based on the build output dir which is not available here
+            const relativeRoutePath = relative(buildDir, `${__dirname}/public-dynamic-config-api-route.js`);
```

Comment:
this is dumb - going to file some issues w/ remix, so we can use a virtual module instead.